### ISSUE TITLE
pylance's inlay hints to pyright?

### ISSUE THREAD

**Issue by FrancescElies** [microsoft/pylance-release#4169]
I am well aware that this question might be closed directly because of this being answered in [faq](https://github.com/microsoft/pylance-release/blob/main/FAQ.md#are-there-any-plans-to-open-source-pylance-in-the-future)

> Because we have plans to include Pylance in Microsoft products outside of VS Code that do not have the same license, business models and/or are closed-source (for example, fully managed products and services), we do not currently have plans to open-source Pylance.
 
> However, a substantial portion of Pylance’s source code is open source via the Pyright type checker. We welcome contributions via that repository, and feedback on Pylance via our pylance-release repo.

I still would like to hear what you think specifically about inlay hints being part of pyright instead of pylance. Lsp support for inlayhints was introduced into the protocol and many implementations out there like [rust-analyzer](https://rust-analyzer.github.io/manual.html) and [typescript-language-server](https://github.com/typescript-language-server/typescript-language-server/blob/70eae7e0885d9b5b7841cad3ba033f3c9c6955d2/README.md#inlay-hints-textdocumentinlayhint) offer inlayhints that can be enabled in any editor supporting lsp.

Vscode is great, I feel making that opensource wouldn't make people go away from vscode into other editors but it feels a bit weird that on one side microsoft's [lsp](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/) open specification has support for inlay hints but the actual implementation for python is locked into a vscode extension instead of being open.

In short, are there any chances we would see microsoft shifting ground and making this part of pyright?

**Comment by rchiodo** [comment]
Sorry I don't think that's likely to happen. These issues are useful in gauging how many people wouldn't use Pylance because it's not open source though.
